### ISSUE TITLE
Unable to use GPU during wav2vec2 decoding 

### ISSUE THREAD

**Issue by manjuke** [huggingface/transformers#21249]
### System Info

Hi All,
I have built a finetuned model for Tamil using facebook/wav2vec2-xls-r-300m. I could do inferencing successfully using CPU. However, Wav2vec2 decoding (with pyctcdecoder) using GPU is not working. I have tried enabling device='gpu' in the decoding script, and also running the decoding script as “python -m torch.distributed.launch --nproc_per_node=<num of GPUs> <Decoding_Script.py>”. But, on monitoring nvidia-smi output during decoding, none of these methods are using GPU for decoding. Pls suggest @sanchit-gandhi . Thanks

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Use 'cuda' as device for decoding

        audio_name=pd.DataFrame(test_data).audio[i]
        text_org = pd.DataFrame(test_data).text[i]
        audio_input, sample_rate = sf.read(audio_name)
        
        #withLM
        input_values = processor_with_lm(audio_input, sampling_rate=16000, return_tensors="pt").input_values
        logits = model(input_values).logits
        hypothesis = processor_with_lm.batch_decode(logits.detach().numpy()).text
        text_with_lm = hypothesis[0]
        #withoutLM
        input_values_wo = processor(audio_input, sampling_rate=16000, return_tensors="pt").input_values
        logits_wo = model(input_values_wo).logits
        predicted_ids = torch.argmax(logits_wo, dim=-1)
        hypothesis_wo_lm = processor.decode(predicted_ids[0])
        text_wo_lm = hypothesis_wo_lm.replace('[PAD]','')
        

### Expected behavior

GPU decoding should have happened and nvidia-smi output to be shown accordingly.

**Comment by OllieBroadhurst** [comment]
`pyctcdecode` doesn't support GPU

**Comment by sanchit-gandhi** [comment]
Indeed, `pyctcdecode` is a CPU only decoding method. PyTorch recently released a fast beam search decoder with a Flashlight backend: https://pytorch.org/blog/fast-beam-search-decoding-in-pytorch-with-torchaudio-and-flashlight-text/

We could look at integrating this into transformers for faster CTC + LM decoding!
